autoscale/src/main/java/com/sequenceiq/periscope/service/NodeDeletionService.java

```diff
@@ -0,0 +1,86 @@
+package com.sequenceiq.periscope.service;
+
+import java.util.List;
+import java.util.Set;
+
+import javax.inject.Inject;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+
+import com.google.common.collect.Lists;
+import com.sequenceiq.cloudbreak.api.endpoint.v4.common.Status;
+import com.sequenceiq.cloudbreak.api.endpoint.v4.stacks.response.StackV4Response;
+import com.sequenceiq.cloudbreak.message.CloudbreakMessagesService;
+import com.sequenceiq.periscope.api.model.ScalingStatus;
+import com.sequenceiq.periscope.common.MessageCode;
+import com.sequenceiq.periscope.domain.Cluster;
+import com.sequenceiq.periscope.domain.LoadAlert;
+import com.sequenceiq.periscope.monitor.handler.CloudbreakCommunicator;
+import com.sequenceiq.periscope.notification.HttpNotificationSender;
+import com.sequenceiq.periscope.utils.StackResponseUtils;
+
+@Service
+public class NodeDeletionService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(NodeDeletionService.class);
+
+    @Inject
+    private CloudbreakCommunicator cloudbreakCommunicator;
+
+    @Inject
+    private StackResponseUtils stackResponseUtils;
+
+    @Inject
+    private HistoryService historyService;
+
+    @Inject
+    private HttpNotificationSender notificationSender;
+
+    @Inject
+    private CloudbreakMessagesService messagesService;
+
+    public void deleteStoppedNodesIfPresent(Cluster cluster) {
+        Set<LoadAlert> loadAlerts = cluster.getLoadAlerts();
+        if (loadAlerts.isEmpty()) {
```

Comment:
Pass in the hostGroup as a parameter. So that we can persist the DB operation, and then invoke the delete.

Reply:
Updated to pass hostGroup as an argument